staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -737,7 +739,15 @@ func (t CRDRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (gen
 		CountMetricPollPeriod:   t.CountMetricPollPeriod,
 	}
 	if t.EnableWatchCache {
-		ret.Decorator = genericregistry.StorageWithCacher(t.DefaultWatchCacheSize)
+		sizes, err := serveroptions.ParseWatchCacheSizes(t.WatchCacheSizes)
+		if err != nil {
+			return generic.RESTOptions{}, err
```

Comment:
When would this happen? If user specify an invalid cache size? Would it make sense to use default cache size to keep backward compatibility?

Reply:
Hi @mbohlool, thanks for your review. This RESTOptions is needed when we initial new storage. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go#L101 
But the value itself is passed through `--watch-catch-size` param during flag parsing.

The backward compatibility is assured because the same behavior can be expected if the user didn't specify any CRD related cache size through the`--watch-catch-size`.

If a user specify an invalid cache size,  `ParseWatchCacheSizes` function will throw an invalid cache size `Error`. This is consistent with the behavior for apiserver. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go#L224

Please let me know if it is better for us to ignore the error and continue to initialize the cache with default size.

Reply:
Why don't we parse earlier so that here we cannot get an error?

Reply:
Not only validate, but actually convert it to the target format very early in the process. Parsing user data late is not a good idea.